packages/engine/src/scene/components/GroundPlaneComponent.ts

```diff
@@ -48,45 +57,38 @@ export const GroundPlaneComponent = defineComponent({
     visible: S.Bool(true)
   }),
 
-  onInit(entity) {
-    return {
-      color: 0xffffff as ColorRepresentation,
-      visible: true
-    }
-  },
-
-  onSet(entity, component, json) {
-    if (!json) return
-
-    if (matchesColor.test(json.color)) component.color.set(json.color)
-    if (matches.boolean.test(json.visible)) component.visible.set(json.visible)
-  },
-
-  toJSON: (component) => {
-    return {
-      color: component.color,
-      visible: component.visible
-    }
-  },
-
   reactor: function () {
     const entity = useEntityContext()
 
     const component = useComponent(entity, GroundPlaneComponent)
 
-    const getMaterial = (): MeshLambertMaterial | ShadowMaterial => {
-      return component.visible.value ? new MeshLambertMaterial() : new ShadowMaterial({ opacity: 0.5 })
+    const meshEntity = useHookstate(() => {
```

Comment:
are we able to keep the mesh entity on the current entity?

Reply:
Yeah ok, I think we should hold off on trying to fix that category of problem since we're going to refactoring that stuff soon, and need a general solution.